### PR TITLE
More features for 2d animations

### PR DESCRIPTION
**Pull request by johnomotani**
Replace `Imshow` with `Pcolormesh` for 2d animations. Add optional arguments for extra features.

Should be reviewed/merged after #71.

### CODE REVIEW

File: xbout/plotting/animate.py [hunk 1]
```diff
@@ -1,21 +1,127 @@
...
+
+    ax.set_title(da.name)
+    ax.set_xlabel(x)
+    ax.set_ylabel(y)
+
+    if separatrix:
+        plot_separatrices(da, ax)
+
+    if targets:
+        plot_targets(da, ax, hatching=add_limiter_hatching)
```
Comment (rdoyle45) [inline xbout/plotting/animate.py]: Should we check if the data is core-only as we do in `plot2d_wrapper`?

```
if _is_core_only(da):
         separatrix = False
         targets = False
 
if separatrix:
         plot_separatrices(da, ax)

if targets:
         plot_targets(da, ax, hatching=add_limiter_hatching)
```

**Comment by TomNicholas** [comment]
Tests are failing here but passing in #73, which is a bit odd because I thought #73 built on this one?

**Comment by rdoyle45** [comment]
> Tests are failing here but passing in #73, which is a bit odd because I thought #73 built on this one?

It does build on this I believe. It is only the tests that use this that fail. 
```
export PIP_PACKAGES="setuptools pip pytest pytest-cov coverage codecov boutdata xarray==0.12.2 dask==1.0.0 natsort==5.5.0 matplotlib==3.1.1 animatplot==0.4.0 netcdf4==1.4.2 Pillow==6.1.0"
```

The tests in #73 don't use this. I can't immediately see why this would give an issue.

**Comment by TomNicholas** [comment]
Could it be https://github.com/t-makaro/animatplot/pull/40 that's needed? One of the differences between the tests is animatplot v0.4.0 vs v0.4.1

**Comment by rdoyle45** [comment]
Happy for this to merge as is @TomNicholas.

**Comment by TomNicholas** [comment]
Thanks @rdoyle45 !

**Pull request merged**